tests/platforms/android/gradle/test_package__aab.py

```diff
@@ -0,0 +1,117 @@
+from subprocess import CalledProcessError
+from unittest.mock import MagicMock
+
+import pytest
+
+from briefcase.exceptions import BriefcaseCommandError
+
+from ....utils import create_file
+
+
+@pytest.fixture
+def first_app_aab(first_app_config):
+    first_app_config.packaging_format = "aab"
+    return first_app_config
+
+
+def test_unsupported_template_version(package_command, first_app_generated):
+    """Error raised if template's target version is not supported."""
+    # Mock the build command previously called
+    create_file(package_command.binary_path(first_app_generated), content="")
+
+    package_command.verify_app = MagicMock(wraps=package_command.verify_app)
+
+    package_command._briefcase_toml.update(
+        {first_app_generated: {"briefcase": {"target_epoch": "0.3.16"}}}
+    )
+
+    with pytest.raises(
+        BriefcaseCommandError,
+        match="The app template used to generate this app is not compatible",
+    ):
+        package_command(first_app_generated, packaging_format="aab")
+
+    package_command.verify_app.assert_called_once_with(first_app_generated)
+
+
+def test_default_packaging_format(package_command):
+    assert package_command.default_packaging_format == "aab"
```

Comment:
This test case looks like it should be in the base test_package; although it's returning `aab`, that's because it's a default, not because aab has been selected here.